ImageD11/sinograms/point_by_point.py

```diff
@@ -1651,42 +1696,30 @@ def idxpoint(
     """
     cImageD11.cimaged11_omp_set_num_threads(1)
     # args    isel, omega, dtyi, gx, gy, gz
+    # ring mask
     mr = isel
-    # select the peaks using the geometry module
+    # mask all the peaks by dtyi and omega (for scoring)
     m = dtyimask_from_sincos(i, j, sinomega, cosomega, dtyi, y0, ystep)
-    # now we correct g-vectors for origin point
+    # combine masks together (for indexing)
+    local_mask = m & mr
 
-    # compute the sample coordinates for this pixel
-    sx, sy = step_to_sample(i, j, ystep)
-    dty = dtyi_to_dty(dtyi, ystep)
-
-    # compute x distance offset for this pixel
-    lx, _ = sample_to_lab_sincos(sx, sy, y0, dty, sinomega, cosomega)
-    parlocal = parglobal.get_parameters().copy()
-    parlocal['distance'] = parlocal['distance'] - lx
-
-    # compute tth eta for all peaks with corrected distance at this pixel
-    tth, eta = ImageD11.transform.compute_tth_eta((sc, fc), **parlocal)
-
-    # compute gvectors for all peaks with corrected distance at this pixel
-    gx, gy, gz = ImageD11.transform.compute_g_vectors(tth, eta,
-                                                      omega,
-                                                      parlocal['wavelength'],
-                                                      wedge=parlocal['wedge'],
-                                                      chi=parlocal['chi'])
-
-    # mask g-vectors to this pixel for indexing
-    inds = np.mgrid[0: len(gx)][m & mr]
-    gv = np.empty((len(inds), 3), "d")
-    gv[:, 0] = gx[inds]
-    gv[:, 1] = gy[inds]
-    gv[:, 2] = gz[inds]
+    # we need to index using peaks[m & mr]
+    # then score with peaks[m]
+    # so we need to compute g-vectors for peaks[m]
+    # then mask those for indexing via [mr]
+
+    # compute g-vectors for peaks[m]
+    gv, gx, gy, gz = get_local_gv(i, j, ystep, omega[m], sinomega[m], cosomega[m], xl[m], yl[m], zl[m])
+    eta_local = eta[m]
+
+    # mask g-vectors to [m & mr]
+    gv_local_mask = gv[local_mask[m], :]
```

Comment:
The comment suggests this should be: `gv[ m & mr , : ]`  or `gv[ local_mask, : ]`. Not sure why there is an [m] on the mask?

Reply:
My thought process went like this:

We want to calculate the g-vectors for all the peaks at this pixel (mask `m`).
However, we want to index only the g-vectors that are also in `isel` (mask `mr`).

We compute `gv` (and `gx, gy, gz`) from `peaks[m]` (the `dtyi, omega` mask) so `len(gv) = m.sum() != len(sinomega)`.

However `mr` has the wrong length to directly mask `gv`.
If we combine `local_mask = m & mr` first, then we can mask `local_mask` by `m` to get the `mr` mask for only the `m`-masked values, so the masks have the right lengths.
Therefore `gv[local_mask[m], :]` should be correct.

Reply:
If you want the ring mask at this pixel I would expect to read that as  `mr[ local_mask ]` ?

Reply:
The way I see it is that we are creating a (ring and pixel) mask for all the peaks using `local_mask = m & mr`, then selecting a (pixel) subset of that with `local_mask[m]` which then has the same length to index `gv` (which was also created from `peaks[m]`).

Reply:
Perhaps the problem is just unclear variable naming?

`m`: Mask on all peaks to select those that come from the pixel
`mr`: Mask on all peaks to select those on the ring (`isel`)
`local_mask`: Mask on all peaks to select those on the ring AND those from the pixel

Reply:
It seems like it can also be `gv[ mr[m] ]` with a comment like `# selection: mr = active on rings, [m] at this point in space`.

The thing you have is:
`gv[ ( m & mr )[m] ]`

While I see this is mathematically equivalent, having the m in there twice meant I could not figure out what you were doing.

Feel free to merge and I can always change it later.